scripts/metric_reporter/reporter/base_reporter.py

```diff
@@ -6,13 +6,27 @@
 
 import csv
 import logging
+import re
+from datetime import datetime
 from pathlib import Path
 from typing import Any, Sequence
 from pydantic import BaseModel
 
-
 DATE_FORMAT = "%Y-%m-%d"
-DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S"  # e.g., 2024-07-10T00:24:56
+DATETIME_REGEX = r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}$"
+UTC_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"  # e.g., 2024-09-04T14:19:43Z
+UTC_DATETIME_REGEX = r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$"
+MICROSECOND_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%f"  # e.g., 2024-09-04T13:49:36.833990
+MICROSECOND_DATETIME_REGEX = r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{1,6}$"
+UTC_MICROSECOND_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"  # e.g., 2024-05-15T14:25:27.188Z
+UTC_MICROSECOND_DATETIME_REGEX = r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{1,6}Z$"
+OFFSET_MICROSECOND_DATETIME_FORMAT = (
+    "%Y-%m-%dT%H:%M:%S.%f%z"  # e.g., 2024-08-28T18:24:29.959+00:00
```

Comment:
Per https://docs.python.org/3/library/datetime.html


Directive | Meaning | Example
-- | -- | --
%z | UTC offset in the form ±HHMM[SS[.ffffff]] (empty string if the object is naive). | (empty), +0000, -0400, +1030, +063415, -030712.345216

Not sure the example matches the docs (specifically the `:` bit). 🤷

Reply:
uhg... you know what... dateutil (ref https://github.com/mozilla/ecosystem-test-scripts/pull/21/commits/37c2f7f5c576fa8ae21a67242f56cb43df9a7612)